definition_validator/validator.cc

```diff
@@ -112,15 +112,34 @@ void matchPositional(const core::Context ctx, absl::InlinedVector<reference_wrap
     }
 }
 
-// Ensure that two argument lists are compatible in shape and type
+// Ensure that two argument lists are compatible in shape and type and that method visibility is compatible
 void validateCompatibleOverride(const core::Context ctx, core::MethodRef superMethod, core::MethodRef method) {
-    if (method.data(ctx)->flags.isOverloaded) {
+    auto methodData = method.data(ctx);
+
+    if (methodData->flags.isOverloaded) {
         // Don't try to check overloaded methods; It's not immediately clear how
         // to match overloads against their superclass definitions. Since we
         // Only permit overloading in the stdlib for now, this is no great loss.
         return;
     }
 
+    auto superMethodData = superMethod.data(ctx);
```

Comment:
Based on the [internals doc](https://github.com/sorbet/sorbet/blob/master/docs/internals.md) my understanding is that the `.data` calls are somewhat slow and I found the lookup repeated in several places (though largely inside conditionals). I figured it might make sense to extract this and replace it below but I can make this change more isolated if preferred.